src/v/kafka/server/group.cc

```diff
@@ -2789,8 +2803,9 @@ group::do_commit(kafka::group_id group_id, model::producer_identity pid) {
 }
 
 void group::abort_old_txes() {
-    ssx::spawn_with_gate(_gate, [this] {
-        return do_abort_old_txes().finally([this] {
+    ssx::spawn_with_gate(_gate, [this]() -> ss::future<> {
+        auto lock = co_await _state_lock.hold_read_lock();
```

Comment:
why does aborting only need a read lock? please explain the locking rules in the commit message.

Reply:
See comment bellow
Add  this comments to code